src/nvim/eval.c

```diff
@@ -16406,11 +16409,28 @@ static void get_system_output_as_rettv(typval_T *argvars, typval_T *rettv,
     return;  // Already did emsg.
   }
 
+  if (p_verbose > 3) {
+    char buf[NUMBUFLEN];
+    verbose_enter();
+    smsg(_("Calling shell to execute: \"%s\""), tv_get_string_buf(argvars, buf));
+    smsg(_("")); // TODO: why do I need to add this?
```

Comment:
For some reason... it doesn't print the first message... but I don't know the code base enough to see if something funny is happening. I hope I have time to investigate that during the weekend.

The result of the functional test without L16416:

```test/functional/ui/screen.lua:307: Row 10 did not match.
Expected:
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |*Calling shell to execute: "echo hi"                  |
  |                                                     |
  |hi                                                   |
  |                                                     |
  |Press ENTER or type command to continue^              |
Actual:
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |~                                                    |
  |*                                                     |
  |                                                     |
  |hi                                                   |
  |                                                     |
  |Press ENTER or type command to continue^              |

To print the expect() call that would assert the current screen state, use
screen:snapshot_util(). In case of non-deterministic failures, use
screen:redraw_debug() to show all intermediate screen states.  
```

Reply:
If I do:

    git grep smsg

I see that it's a common pattern, e.g.:

```c
    smsg(_("Exception caught: %s"), excp->value);
    msg_puts("\n");  // don't overwrite this either
```

So I think it's fine as you have it. Though perhaps change it to `msg_puts("\n"); `.